src/rust/engine/src/nodes.rs

```diff
@@ -1144,10 +1127,18 @@ impl NodeKey {
   ///
   pub fn fs_subject(&self) -> Option<&Path> {
     match self {
+      &NodeKey::DigestFile(ref s) => Some(s.0.path.as_path()),
       &NodeKey::ReadLink(ref s) => Some((s.0).0.as_path()),
       &NodeKey::Scandir(ref s) => Some((s.0).0.as_path()),
       &NodeKey::Stat(ref s) => Some(s.0.as_path()),
```

Comment:
I believe that after this review `Stat` should be unused, since `FileDigest` is the only thing representing presence of a file in a `Snapshot`? See the comment on the `Stat` struct.

Reply:
Nice! Done! Hurrah for replacing weird placeholders which need long comments with things which actually make standalone sense!